ktor-features/ktor-i18n/jvm/test/io/ktor/tests/i18n/TranslationTest.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.tests.i18n
+
+import io.ktor.application.*
+import io.ktor.http.*
+import io.ktor.http.HttpStatusCode.Companion.OK
+import io.ktor.routing.*
+import io.ktor.server.testing.*
+import kotlin.test.*
+import io.ktor.i18n.*
+import io.ktor.response.*
+
+class TranslationTest {
+
+    @Test
+    fun testTranslationToLanguageSpecifiedInAcceptLanguageHeader() = withTestApplication {
+        application.install(I18n) {
+            encoding = java.nio.charset.StandardCharsets.UTF_8
+        }
+
+        application.routing {
+            get("/") {
+                val valueInPortuguese = call.translate("some_key")
+                call.respond(OK, valueInPortuguese)
+            }
+        }
+
+        handleRequest(HttpMethod.Get, "/") {
+            addHeader(HttpHeaders.AcceptLanguage, "pt-BR")
+        }.response.let { response ->
+            assertEquals(OK, response.status())
+            assertNotNull(response.content)
+
+            val contentAsString = response.content!!
+            assertEquals("Portuguese Key", contentAsString)
+        }
+    }
+
+    @Test
+    fun testTranslationToDefaultLanguage() = withTestApplication {
+        application.install(I18n) {
+            defaultLanguage = "en-US"
+        }
+
+        application.routing {
+            get("/") {
+                val valueInPortuguese = call.translate("some_key")
+                call.respond(OK, valueInPortuguese)
+            }
+        }
+
+        handleRequest(HttpMethod.Get, "/").response.let { response ->
+            assertEquals(OK, response.status())
+            assertNotNull(response.content)
+
+            val contentAsString = response.content!!
+            assertEquals("English Key", contentAsString)
+        }
+    }
+
+    @Test
+    fun testDefaultBundleWhenDefaultLanguageIsNotConfigured() = withTestApplication {
+        application.install(I18n)
+
+        application.routing {
+            get("/") {
+                val valueInPortuguese = call.translate("some_key")
+                call.respond(OK, valueInPortuguese)
+            }
+        }
+
+        handleRequest(HttpMethod.Get, "/").response.let { response ->
+            assertEquals(OK, response.status())
+            assertNotNull(response.content)
+
+            val contentAsString = response.content!!
+            assertEquals("Default key", contentAsString)
+        }
+    }
+
```

Comment:
We also need a test for a case when wrong or corrupted accepted language provided by a client. I believe we need to fall back to the default language in this case.

Also, it is a common case when multiple languages are provided in `accept-language` header. Some of them could be illegal. For example, my personal header generated by my browser is `ru-RU,ru;q=0.8,sv;q=0.5,en;q=0.3`

Reply:
Just added this test case, with almost exactly this example :)